line-bot-api-client/src/main/java/com/linecorp/bot/client/LineMessagingClientImpl.java

```diff
@@ -75,22 +76,42 @@
 
     @Override
     public CompletableFuture<BotApiResponse> pushMessage(final PushMessage pushMessage) {
-        return toBotApiResponseFuture(retrofitImpl.pushMessage(pushMessage));
+        return toBotApiResponseFuture(retrofitImpl.pushMessage(null, pushMessage));
+    }
+
+    @Override
+    public CompletableFuture<BotApiResponse> pushMessage(final UUID retryKey, final PushMessage pushMessage) {
+        return toBotApiResponseFuture(retrofitImpl.pushMessage(retryKey.toString(), pushMessage));
     }
 
     @Override
     public CompletableFuture<BotApiResponse> multicast(final Multicast multicast) {
-        return toBotApiResponseFuture(retrofitImpl.multicast(multicast));
+        return toBotApiResponseFuture(retrofitImpl.multicast(null, multicast));
+    }
+
+    @Override
+    public CompletableFuture<BotApiResponse> multicast(final UUID retryKey, final Multicast multicast) {
+        return toBotApiResponseFuture(retrofitImpl.multicast(retryKey.toString(), multicast));
     }
 
     @Override
     public CompletableFuture<BotApiResponse> broadcast(Broadcast broadcast) {
-        return toBotApiResponseFuture(retrofitImpl.broadcast(broadcast));
+        return toBotApiResponseFuture(retrofitImpl.broadcast(null, broadcast));
```

Comment:
```suggestion
        return broadcast(null, broadcast);
```

Calling `multicast(final UUID retryKey, final Multicast multicast)` (delegation) is better than calling `toBotApiResponseFuture` and `retrofitImpl.broadcast` here.

Reply:
If I use delegation, I need to put `@Nullable` annotation to `broadcast(final UUID retryKey, final Broadcast broadcast)`. It's not a good API.

Reply:
Aha, you are calling `retryKey.toString()` your self. OK.